src/jit/emitxarch.cpp

```diff
@@ -10724,7 +10774,6 @@ BYTE* emitter::emitOutputRI(BYTE* dst, instrDesc* id)
         // Handle SSE2 instructions of the form "opcode reg, immed8"
 
         assert(id->idGCref() == GCT_NONE);
-        assert(valInByte);
```

Comment:
Removed an assertion. This did not work with the jump-table fallback "opcode reg, (128-255)".

Reply:
That seems strange, I would expect that 128-255 would qualify as "fits in byte"

Reply:
@CarolEidt, do you have any context as to why `fitsInByte` is really `fitsInSByte`?

Reply:
Seems we cannot directly remove this assertion. On 32-bit platform (x86 build), an strange assertion failure on `vpslldq  xmm0, 128`
```
Assert failure(PID 96 [0x00000060], Thread: 10976 [0x2ae0]): Assertion failed 'sz == _idCodeSize' in 'System.Runtime.Intrinsics.X86.Sse2:ShiftLeftLogical128BitLane(struct,ubyte):struct' (IL size 8)

    File: d:\workspace\coreclr-imm\src\jit\emit.h Line: 997
    Image: D:\workspace\coreclr-imm\bin\tests\Windows_NT.x86.Debug\Tests\Core_Root\CoreRun.exe
```

Reply:
I realized this is just an issue with casting.

We are taking an `unsigned int8` and converting it to a `signed native int`, which causes zero extension.

We need to first cast to `signed int8` so that we wind up with a signed extension instead.